src/ILLink.Tasks/LinkTask.cs

```diff
@@ -67,6 +67,18 @@ public class ILLink : ToolTask
 		/// </summary>
 		public string Warn { get; set; }
 
+		/// <summary>
+		/// The list of warnings to report as errors.
+		/// Maps to '--warnaserror'.
+		/// </summary>
+		public string WarningsAsErrors { get; set; }
+
+		/// <summary>
+		/// The list of warnings to report as usual.
+		/// Maps to '--warnaserror-'.
+		/// </summary>
+		public string WarningsNotAsErrors { get; set; }
+
```

Comment:
Would you be able to add a boolean `TreatWarningsAsErrors` for the global setting? This should go first on the command-line, so that the individual settings take priority.